code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -52,6 +52,13 @@
 	if(board)
 		piping_layer = board.pipe_layer
 		set_layer = piping_layer
+
+	for(var/obj/machinery/atmospherics/device in get_turf(src))
+		if(device.piping_layer != piping_layer || device == src)
+			continue
+		visible_message("<span class='warning'>Something is hogging the pipe, remove the obstruction or change the machine piping layer.</span>")
```

Comment:
This seems like it wouldn't make a whole lot of sense in game, especially if (worst case senario) they're trying to build a thermomachine on a floor tile covering a pipe

Reply:
Why not? The pipe it's talking about is the thermomachine one, how should I phrase it?

Reply:
done